ocr-gcp-cloud-vision/python/services/ocr.py

```diff
@@ -0,0 +1,256 @@
+# which the users could be uploading.
+# A user can upload only one photo, as of this demo.
+# `reading` and `writing` permissions should be provided
+# to this Python application for performing read/write
+# operations on the Storage Bucket.
+photo_bucket = bucket("photos")
+photos = photo_bucket.allow("reading", "writing")
+
+
+@ocr_api.post("/profiles")
+async def create_user_profile(ctx: HttpContext) -> None:
+    """
+    A POST API to create a user's profile and store
+    it in a Key-Value store.
+
+    Args:
+        ctx (HttpContext): HTTP Request and Response context for the API.
+    """
+    # Generate a unique profile id for the user.
+    user_profile_id = str(uuid4())
+
+    # Initialize a set of JSON keys which should be present in the
+    # HTTP Request Payload.
+    required_json_payload_keys = {"name", "age", "city"}
+
+    # If HTTP Request JSON Payload is not empty, proceed with user profile
+    # creation, else return an error.
+    if ctx.req.json is not None:
+        # Check the difference between the desired HTTP Request JSON Payload
+        # keys and the ones sent over the actual HTTP Request.
+        missing_json_payload_keys = required_json_payload_keys - set(ctx.req.json)
+
+        # If any required keys are missing in the HTTP Request JSON Payload,
+        # return an HTTP 400 Bad Request response.
+        if missing_json_payload_keys:
+            ctx.res.status = 400
+            ctx.res.body = {
+                "msg": f"Bad Request. Missing required keys: {missing_json_payload_keys}."
+            }
+            # Return `None` to break the flow and return the intended response.
+            return
+    else:
+        # Return an HTTP 400 Bad Request response in case the HTTP Request JSON Payload
+        # is empty.
+        ctx.res.status = 400
+        ctx.res.body = {
+            "msg": f"Bad Request. Missing required keys: {required_json_payload_keys}."
+        }
+        # Return `None` to break the flow and return the intended response.
+        return
+
+    # Store the user profile details in the Key-Value Store.
+    # It should be noted that a Python dictionary is expected as the value for the key.
+    # Therefore, when retrieving the data from the Key-Value Store, if JSON processing needs to
+    # be done, a `json.dumps(data)` would help.
+    try:
+        await user_profiles_kv.set(
+            user_profile_id,
+            {
+                "name": ctx.req.json["name"],
+                "age": ctx.req.json["age"],
+                "city": ctx.req.json["city"],
+            },
+        )
+
+        # Return an HTTP 200 OK message stating that the creation of the user profile
+        # happened successfully.
+        ctx.res.body = {
+            "msg": f"Profile with id '{user_profile_id}' created successfully."
+        }
+        # Return `None` to break the flow and return the intended response.
+        return
+    except Exception:
```

Comment:
It could be worth noting that the Python SDK will catch any unhandled exceptions and return a internal server error by default. This avoids the risk that an exception accidentally leaks sensitive data to a client.